doc/dev/backend-plugins.md

```diff
@@ -130,7 +130,7 @@ It is where validation, generation of implicit values and similar tasks happen.
 
 Finally, `libelektra-kdb` merges the keyset returned by the `poststorage` phase with the ones returned by other backend plugins for different mountpoints and then returns it to the user.
 
-TODO: Are modifications to `parentKey` visible to the user?
+<!-- TODO: Should modifications to `parentKey` (beyond errors/warnings) be visible to the user? -->
```

Comment:
input welcome

Reply:
Yes, everything should be visible to the user. The string is by convention the "connection" to the underlying storage (file). But @flo91 can reconsider this. Anyway, it should be possible to pass out information by the backends.

Reply:
Ah yes, for things like `kdb file` you need the parentKey value of the backend. However, I think it might be better, if communicate this via metakeys e.g. `meta:/backends/#/storageid`. This would allow passing back the information about _all_ the backends that have been used.

> it should be possible to pass out information by the backends.

Do we need any information beyond that? My metadata idea would allow for limited extra info.

In any case, I think everything should be standardized and backends should be allowed to do whatever to the parentKey.

Reply:
> Ah yes, for things like kdb file you need the parentKey value of the backend. However, I think it might be better, if communicate this via metakeys e.g. meta:/backends/#/storageid. This would allow passing back the information about all the backends that have been used.

I don't see an immediate benefit of it. `kdb file`, the only customer of that feature, would get more complicated.

> In any case, I think everything should be standardized and backends should be allowed to do whatever to the parentKey.

I fully agree. Please document this.

Reply:
> > In any case, I think everything should be standardized and backends should be allowed to do whatever to the parentKey.
> 
> I fully agree. Please document this.

Oops that was a bad typo on my part... I meant to say "backends should **NOT** be allowed to do whatever to the parentKey"

> I don't see an immediate benefit of it.

It would be a bit more structured than the current solution, which seems a bit arbitrary. Especially, if you don't know about the internal behaviour of `kdbGet` and backend plugins.

> `kdb file`, the only customer of that feature, would get more complicated.

I thought `kdb file` could just print all the backends that were accessed, but I guess that's compatible with the current behaviour.

So probably we just keep the current behaviour. Adding a way to find all the backends that where accessed (both parent keys and filenames/storageids) could still be useful for diagnostic purposes. Currently, only `kdbGet` has this information. The caller doesn't even get to know whether the cache was used.